crates/extricrate/src/lib.rs

```diff
@@ -26,6 +28,46 @@ pub mod dependencies {
             Self(value.to_owned())
         }
     }
+    impl From<&ModulePath> for ModuleName {
+        fn from(value: &ModulePath) -> Self {
+            Self(format!("crate::{}", value.0.replace(".", "::")))
+        }
+    }
+
+    impl fmt::Display for ModuleName {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            f.write_str(&self.0)
+        }
+    }
+
+    #[derive(Debug, PartialEq, Eq)]
+    pub struct ModuleList(pub Vec<ModuleName>);
+    impl fmt::Display for ModuleList {
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            f.write_fmt(core::format_args!(
+                "{}",
+                self.0
+                    .iter()
+                    .map(|m| m.to_string())
+                    .collect::<Vec<_>>()
```

Comment:
Wait, didn't we already import Itertools somewhere to avoid having to collect to a vec before joining?